portfolio/src/main/java/com/google/sps/servlets/UserSubmittedCommentsServlet.java

```diff
@@ -32,40 +32,59 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-@WebServlet("/map")
-public class MapServlet extends HttpServlet {
+@WebServlet("/submitted-comments")
+public class UserSubmittedCommentsServlet extends HttpServlet {
   private static final String NAME = "name";
   private static final String PHONE = "phone";
   private static final String LOCATION = "location";
   private static final String DESCRIPTION = "description";
   private static final String TABLE_NAME = "Pins";
   private static final String TIME_STAMP = "timestamp";
+  private static final String ID = "id";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    Query query = new Query(TABLE_NAME);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<UserComment> userComments = new ArrayList<UserComment>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty(NAME);
+      String phone = (String) entity.getProperty(PHONE);
+      String location = (String) entity.getProperty(LOCATION);
+      String description = (String) entity.getProperty(DESCRIPTION);
+      long timeStamp = (long) entity.getProperty(TIME_STAMP);
+      long id = entity.getKey().getId();
```

Comment:
Do you need to pass the id back to user?

Reply:
Not necessarily, but I save the ID to the UserComment object just in case I will need to retrieve a specific comment in the future.